hack/release-helper.sh

```diff
@@ -0,0 +1,247 @@
+                operator_tag=${OPTARG} ;;
+            e)
+                enclave_cc_tag=${OPTARG} ;;
+            k)
+                kata_tag=${OPTARG} ;;
+            *)
+                usage_and_exit ;;
+        esac
+    done
+
+    # minimal checking: ensure all args are provided when using 'update'
+    if [ ${action} = "update" ]; then
+        if [[ ! -v operator_tag ]] || [[ ! -v enclave_cc_tag ]] || [[ ! -v kata_tag ]]; then
+            echo "Error: Missing ARGS"
+            usage_and_exit
+        fi
+    fi
+}
+
+
+#
+# Make sure that the pre-reqs container ships the same version of the nydus
+# snapshotter that we use in the kata ci. Update the Makefile if needed.
+#
+function check_nydus() {
+    echo "Checking that the operator's nydus version matches the one used in kata"
+
+    kata_versions="https://raw.githubusercontent.com/kata-containers/kata-containers/refs/tags/${kata_tag}/versions.yaml"
+    kata_versions_dest="versions.yaml.$$"
+
+    # pull the versions.yaml that kata uses
+    wget --quiet "${kata_versions}" -O "${kata_versions_dest}"
+
+    # multi-line grep to find the nydus-snapshotter version in kata's version.yaml
+    # then awk/tail/tr and string manipulation to sand it down.
+    nydus_version_kata="$(grep -zoP "nydus-snapshotter.*\n.*description.*\n.*url.*\n.*version.*" ${kata_versions_dest} | awk '{print $2}' | tail -n1 | tr -d '\0')"
+    nydus_version_kata=${nydus_version_kata//\"} # strip double quotes
+
+    # delete temporary kata versions.yaml that we downloaded
+    rm "${kata_versions_dest}"
+
+    # grab the nydus version in the operator
+    prereqs_makefile="install/pre-install-payload/Makefile"
+    nydus_version_prereqs="$(grep '^NYDUS_SNAPSHOTTER_VERSION = ' $prereqs_makefile | awk '{ print $3}')"
+
+    # check for match and update if needed
+    if [ "${nydus_version_kata}" == "${nydus_version_prereqs}" ]; then
+        echo "-> OK, no changes made: nydus-snapshotter already up-to-date" \
+             "(version: $nydus_version_prereqs)"
+    else
+        echo "-> Found mismatch: updating nydus-snapshotter from" \
+             "$nydus_version_prereqs to $nydus_version_kata"
+        sed -i "s/^NYDUS_SNAPSHOTTER_VERSION = .*/NYDUS_SNAPSHOTTER_VERSION = $nydus_version_kata/" "${prereqs_makefile}"
+        echo "-> Changes made to ${prereqs_makefile}."
+        echo "   Please verify and open PR as needed."
+    fi
+}
+
+
+#
+# Update the operator tag for the new release.
+#
+function update_operator() {
+    operator_yaml="config/release/kustomization.yaml"
+    echo "Updating operator version to ${operator_tag}"
+    echo "-> Updating ${operator_yaml}"
+    sed -i 's/newTag:.*/newTag: '${operator_tag}'/' ${operator_yaml}
+}
+
+
+#
+# Update the CRDs to point to the most recent version of the pre-reqs payload.
+#
+function update_prereqs() {
+    # grab the hash for the latest commit on the pre-install-payload folder
+    # in github (which requires figuring out the hash of the merge)
```

Comment:
Could you please document in code how this works rather than a simple comment that knowing the latest commit helps? IIUC you're looking for the oldest merge commit that contains the latest change to the pre-install-payload, am I right?

Reply:
Yes, I added a more verbose comment above the function. I'd be grateful for a cleaner way to do this, too, if you see a way.